### PR TITLE
[CPU] Empty tensor used with a custom op leads to CPU plugin exception.

### PR DESCRIPTION
**Pull request by nshchego**
### Details:
 - *Reference node was not executed if one of input tensor was empty.*

### Tickets:
 - *113021*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@maxnick Please take a look

File: src/plugins/intel_cpu/src/nodes/reference.cpp [hunk 1]
```diff
@@ -136,7 +136,11 @@ ov::TensorVector Reference::prepareOutputs() const {
         void *dstDataPtr = getChildEdgesAtPort(i)[0]->getMemory().getData();
         ov::Shape shape = ngraphOp->get_output_partial_shape(i).rank().get_length() == 0 ?
                 ov::Shape{} : getChildEdgesAtPort(i)[0]->getMemory().getStaticDims();
-        outputs.push_back(ov::Tensor(ngraphOp->get_output_element_type(i), shape, dstDataPtr));
+        if (dstDataPtr != nullptr) {
+            outputs.push_back(ov::Tensor(ngraphOp->get_output_element_type(i), shape, dstDataPtr));
+        } else {
+            outputs.push_back(ov::Tensor(ngraphOp->get_output_element_type(i), shape));
+        }
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/reference.cpp:143]: Isn't it more reliable to check the shape on presence of zeroes? There can be other reasons (and not all of them are legit) `dstDataPtr` being `nullptr`, but only in the context of an empty tensor, it's ok to allocate tensors on the ngraph side. Also what about inputs?

File: src/plugins/intel_cpu/src/nodes/reference.cpp [hunk 2]
```diff
@@ -108,22 +108,38 @@ bool Reference::needShapeInfer() const {
...
         ov::Shape shape = ovCoreNode->get_input_partial_shape(i).rank().get_length() == 0 ?
                 ov::Shape{} : getParentEdgesAtPort(i)[0]->getMemory().getStaticDims();
-        inputs.push_back(ov::Tensor(ovCoreNode->get_input_element_type(i), shape, srcDataPtr));
+
+        if (std::any_of(shape.begin(), shape.end(), [](const size_t dim) { return dim == 0lu; } )) {
+            inputs.push_back(ov::Tensor(ovCoreNode->get_input_element_type(i), shape));
+        } else {
+            if (!srcDataPtr) {
+                THROW_CPU_NODE_ERR("has empty input data on port ", i);
+            }
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/reference.cpp:121]: NIT. This is rather an assert construction. Should we introduce `CPU_NODE_ASSERT`?

**Review by maxnick — APPROVED** [review]

File: src/plugins/intel_cpu/src/nodes/reference.cpp (hunk 2 above)
Comment (nshchego) [reply src/plugins/intel_cpu/src/nodes/reference.cpp]: Done

**Comment by slyalin** [comment]
Any progress?

**Pull request merged**